src/parquet/file/metadata.cc

```diff
@@ -485,16 +485,17 @@ class ColumnChunkMetaDataBuilder::ColumnChunkMetaDataBuilderImpl {
     column_chunk_->meta_data.__set_total_uncompressed_size(uncompressed_size);
     column_chunk_->meta_data.__set_total_compressed_size(compressed_size);
     std::vector<format::Encoding::type> thrift_encodings;
-    thrift_encodings.push_back(ToThrift(Encoding::RLE));
     if (has_dictionary) {
-      thrift_encodings.push_back(ToThrift(properties_->dictionary_page_encoding()));
-      // add the encoding only if it is unique
-      if (properties_->version() == ParquetVersion::PARQUET_2_0) {
-        thrift_encodings.push_back(ToThrift(properties_->dictionary_index_encoding()));
+      thrift_encodings.push_back(ToThrift(properties_->dictionary_index_encoding()));
+      if (properties_->version() == ParquetVersion::PARQUET_1_0) {
```

Comment:
The problem is that `dictionary_page_encoding()` returns `PLAIN_DICTIONARY` for V1 files: https://github.com/apache/parquet-cpp/blob/master/src/parquet/column/properties.h#L325

Since Impala writes `Encoding::PLAIN` in the metadata even when dictionary-encoded, I don't want to risk incompatibilities